src/test/java/uk/gov/pay/adminusers/resources/InviteResourceOtpTest.java

```diff
@@ -2,58 +2,218 @@
+                .put("telephone_number", TELEPHONE_NUMBER)
+                .put("password", PASSWORD)
                 .build();
 
         givenSetup()
                 .when()
                 .body(mapper.writeValueAsString(invitationRequest))
                 .contentType(ContentType.JSON)
-                .post(format(INVITES_OTP_RESOURCE_URL, "not-existing-code"))
+                .post(INVITES_GENERATE_OTP_RESOURCE_URL)
+                .then()
+                .statusCode(BAD_REQUEST.getStatusCode());
+    }
+
+    @Test
+    public void validateOtp_shouldCreateUserWhenValidOtp() throws Exception {
+
+        // create an invitation
+        code = InviteDbFixture.inviteDbFixture(databaseHelper)
+                .withEmail(EMAIL)
+                .withOtpKey(OTP_KEY)
+                .withTelephoneNumber(TELEPHONE_NUMBER)
+                .withPassword(PASSWORD)
+                .expired()
+                .insertInvite();
+
+        // generate valid invitationOtpRequest and execute it
+        ImmutableMap<Object, Object> invitationOtpRequest = ImmutableMap.builder()
+                .put("code", code)
+                .put("otp", PASSCODE)
+                .build();
+
+        assertThat(databaseHelper.findInviteByCode(code).size(), is(1));
+
+        ValidatableResponse response = givenSetup()
+                .when()
+                .body(mapper.writeValueAsString(invitationOtpRequest))
+                .contentType(JSON)
+                .accept(JSON)
+                .post(INVITES_VALIDATE_OTP_RESOURCE_URL)
                 .then()
-                .statusCode(NOT_FOUND.getStatusCode());
+                .statusCode(CREATED.getStatusCode());
+
+        String externalId = response.extract().path("external_id");
+
+        // check the response
+        response
+                .statusCode(201)
+                .body("id", nullValue())
+                .body("external_id", is(externalId))
+                .body("username", is(EMAIL))
+                .body("password", nullValue())
+                .body("email", is(EMAIL))
+                .body("telephone_number", is(TELEPHONE_NUMBER))
+                .body("otp_key", is(OTP_KEY))
+                .body("login_counter", is(0))
+                .body("disabled", is(false))
+                .body("_links", hasSize(1))
+                .body("_links[0].href", is("http://localhost:8080/v1/api/users/" + externalId))
+                .body("_links[0].method", is("GET"))
+                .body("_links[0].rel", is("self"));
+
+        // check if the user has been created and it is not disabled in the database
+        List<Map<String, Object>> users = databaseHelper.findUserByUsername(EMAIL);
+
+        assertThat(users.size(), is(1));
+
+        Map<String, Object> createdUser = users.get(0);
+        assertThat(createdUser.get("username"), is(EMAIL));
+        assertThat(createdUser.get("email"), is(EMAIL));
+        assertThat(createdUser.get("password"), is(PASSWORD));
```

Comment:
Instead of creating a UserEntity directly from InviteEntity .. i would rather create a User.java from the invite and pass it to UserService to save (so that it could do the relevant validations and encryptions there)